src/layout/BreadCrumbs.vue

```diff
@@ -331,6 +339,46 @@ export default class BreadCrumbs extends Vue {
     }
   }
 
+  @Watch("configurationResource", { immediate: true })
+  onConfigurationResourceChanged(resource: any) {
+    const configurationItem = this.items.find(
+      (item) => item.title === "Collection:",
+    );
+    if (!configurationItem) return;
+
+    // Trigger fetch when undefined OR null
+    if (resource == null && this.currentConfigurationId) {
+      this.girderResources.forceFetchResource({
+        id: this.currentConfigurationId,
+        type: "upenn_collection",
+      });
+      return;
+    }
+
+    if (resource && resource.name) {
+      Vue.set(configurationItem, "text", resource.name);
+    }
+  }
+
+  @Watch("datasetResource", { immediate: true })
+  onDatasetResourceChanged(resource: any) {
+    const datasetItem = this.items.find((item) => item.title === "Dataset:");
+    if (!datasetItem) return;
+
+    // Trigger fetch when undefined OR null
+    if (resource == null && this.currentDatasetId) {
+      this.girderResources.forceFetchResource({
+        id: this.currentDatasetId,
+        type: "folder",
+      });
+      return;
+    }
+
+    if (resource && resource.name) {
```

Comment:
You can use the `?.` operator to access the property of a potentially undefined value